### PR TITLE
Update 

### PR DESCRIPTION
**Pull request by developer9998**
Recently decided to go back through some of the older versions of the mod, and looking through them I found tons of content that have either stayed the exact same since the release (Mainly through 2021) or have been replaced which content which many others consider as lower quality or worse in a way. With this I've went out my way to edit some of the content I mentioned prior which includes:
- Reverting the Computer model (More space, easier to work with aspect ratio, non-bulky, etc.)
- Changing much of the Game Settings tabs (Mic, Voice, etc.)
- More side changes

Also due to these changes I've bumped the version to 1.6.0, but you can change this and obviously any of this if you want.

## Computer
<br>
- Changed the model (As mentioned)
- Changed key press colours for custom keys (Back, Up, Down, Left, Right)
- Made interacting with the keyboard much easier

## Room tab
<br>
- Made text show for private rooms and group joins

## Name tab
<br>
- Made text show for when you get an error (Empty name, blank name, name too long, etc.)
- Removed spaces in output name when set (Seen in image)

## Turn tab
<br>
- Reorganized page

## Mic tab
<br>
- Reorganized page

## Group tab
<br>
- <b>Actually fixed group joining, pretty sure it's been broken for months</b>
- Reorganized page

## Voice tab
<br>
- Reorganized page

## Items tab (WAS Visuals tab)
<br>
- Reorganized page

## Credits tab
<br>
- Added an actual page system
- Made page sync up with the base game computer

## NEW! Support tab
<br>

## Other changes
- Made Details page also display "User bans yesterday"
- Made Commandline view display "Commandline" instead of "Monkeshell"

### CODE REVIEW

**Comment by developer9998** [comment]
Bit of an unexpected update, but I got some more work done on it. Note the attached images may not be 100% as they appear ingame.

**Comment by gizmogoat** [comment]
I prefer the old monitor
not nostalgia bias btw, I've never used the older versions of CI since I originally used mods on Quest then started playing on PC many many months later once I moved away from integrated graphics

**Comment by developer9998** [comment]
Latest commit fixes computer not spawning in Forest, and places the computer in the new Beach map

**Comment by beanz00** [comment]
i know this isnt the right place to ask but it is the best place i could find, how do i install this if i can? i want to use it but i don't know how to compile it into a mod that works.

**Comment by developer9998** [comment]
Actual question I don't want ignored, is anything going to come out of this or is it just going to be the "Only known method of getting fixed ComputerInterface"

**Comment by Graicc** [comment]
I'd like to merge this eventually but there's a lot of things that need to be changed (by me probably) before its in a mergeable state, so its going to take a while to do all of that work

**Comment by developer9998** [comment]
Alright, fair enough 👍

File: ComputerInterface/QueueManager.cs [hunk 1]
```diff
@@ -28,19 +25,19 @@ public static IQueueInfo GetQueue()
...
+        {
             string currentQueue = PlayerPrefs.GetString("currentQueue", "DEFAULT");
-            if (currentQueue != "DEFAULT" && currentQueue != "COMPETITIVE" && currentQueue != "MINIGAMES")
+            /* if (currentQueue != "DEFAULT" && currentQueue != "COMPETITIVE" && currentQueue != "MINIGAMES")
             {
                 PlayerPrefs.SetString("currentQueue", "DEFAULT");
                 PlayerPrefs.Save();
                 currentQueue = "DEFAULT";
-            }
+            }*/ // most likely the best idea as of right now just in case it's ignored again
```
Comment (Graicc) [inline ComputerInterface/QueueManager.cs]: Was there any bugs this fixed? or just future proofing?

Comment (developer9998) [reply ComputerInterface/QueueManager.cs]: I primarily did it mainly because the last time gamemodes were touched to add Minigames queue this part was forgotten + I'm pretty sure there's an entire interface just for adding custom queues here

File: ComputerInterface/Views/GameSettings/CreditsView.cs
Comment (Graicc) [inline ComputerInterface/Views/GameSettings/CreditsView.cs:1]: You should be able to delegate all of this to the in game credits view, just set the page size to the right number and query GetPage for whatever page

**Comment by developer9998** [comment]
> there's a lot of things that need to be changed (by me probably) before its in a mergeable state

What do you mean by this exactly?

File: ComputerInterface/BaseGameInterface.cs [hunk 2]
```diff
@@ -44,51 +42,113 @@ public static Color GetColor()
...
+                    PlayerPrefs.SetString("playerName", name);
+                    PlayerPrefs.Save();
+
+                    GetColor(out var r, out var g, out var b);
+                    InitializeNoobMaterial(r, g, b);
+                }
+                catch
+                {
+                    throw new NullReferenceException(); // usually occurs due to the InitializeNoobMaterial method
+                }
```
Comment (Graicc) [inline ComputerInterface/BaseGameInterface.cs]: Since another exception is immediately thrown, does this even do anything?

Comment (developer9998) [reply ComputerInterface/BaseGameInterface.cs]: Ah that, I forgot to remove it when I did fix the method mentioned in the comment

Comment (developer9998) [reply ComputerInterface/BaseGameInterface.cs]: Usually when it would falsely check to see if it was or wasn't null I don't remember

**Comment by developer9998** [comment]
Oh neat

**Comment by Graicc** [comment]
> > there's a lot of things that need to be changed (by me probably) before its in a mergeable state
> 
> What do you mean by this exactly?

Things like the above changes, there shouldn't be a singleton for CustomScreenInfo (or really anywhere in CI, that's what Zenject / Dependency Injection is all about), credits view should just ask the base game for the current page, there's some files with only whitespace changes, and presumably other stuff that I don't know about yet

File: ComputerInterface/Patches/InitializeGamemodePatch.cs [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 namespace ComputerInterface.Patches
 {
+    // why is this on computer interface? it feels more like a utilla feature
```
Comment (Graicc) [inline ComputerInterface/Patches/InitializeGamemodePatch.cs]: Computer interface was doing stuff with modifing gamemodes first
```suggestion
```

Comment (developer9998) [reply ComputerInterface/Patches/InitializeGamemodePatch.cs]: Ah makes sense

File: ComputerInterface/Views/GameSettings/ItemSettingsView.cs [hunk 4]
```diff
@@ -24,54 +23,51 @@ public override void OnShow(object[] args)
             Redraw();
         }
 
-		void UpdateState()
-		{
-			_selectionHandler.CurrentSelectionIndex = BaseGameInterface.GetItemMode() ? 1 : 0;
-            _insVolumeFloat = BaseGameInterface.GetInstrumentVolume();
-		}
+		void UpdateState() => _insVolumeFloat = BaseGameInterface.GetInstrumentVolume();
```
Comment (Graicc) [inline ComputerInterface/Views/GameSettings/ItemSettingsView.cs]: Getting rid of the first line prevents the selection from being set to the correct value when the view is first shown. It will always show `Enabled`, even if it is `Disabled`.

**Comment by Graicc** [comment]
Room browser isn't possible anymore because it relies on photon features that no longer exist in Gorilla Tag

**Comment by developer9998** [comment]
Rip room browser

File: ComputerInterface/CustomKeyboardKey.cs [hunk 5]
```diff
@@ -74,61 +79,71 @@ public void Init(CustomComputer computer, EKeyboardKey key, Text keyboardText, s
...
-            var pos = transform.localPosition;
-            pos.y -= KEY_BUMP_AMOUNT;
-            transform.localPosition = pos;
+            if (!_bumped)
+            {
+                _bumped = true;
+                var pos = transform.localPosition;
+                pos.y -= KEY_BUMP_AMOUNT;
+                transform.localPosition = pos;
+                collider.center -= new Vector3(0, 0, KEY_BUMP_AMOUNT / 1.125f);
```
Comment (Graicc) [inline ComputerInterface/CustomKeyboardKey.cs:129]: Why are you dividing by 1.125? I'm assuming its related to the local scale but I didn't notice any differences after removing it

Comment (developer9998) [reply ComputerInterface/CustomKeyboardKey.cs:129]: From adjusting a Unity project of the game, I felt like that change made for easier typing as that from the testing I did essentially made the collider stay in the same location whenever you pressed a key. This change was pre-beachmap update and I know buttons and probably keys were adjusted in some ways (Which includes Forest/Stump computer)

File: ComputerInterface/Views/GameSettings/JoinRoomView.cs [hunk 6]
```diff
@@ -16,23 +12,25 @@ namespace ComputerInterface.Views.GameSettings
...
         {
             base.OnShow(args);
+
+            callbacks = new GameObject();
+            callbacks.name = "RoomCallbacks";
+            UnityEngine.Object.DontDestroyOnLoad(callbacks);
+
+            JoinRoomViewCallbacks calllbacksComponent = callbacks.AddComponent<JoinRoomViewCallbacks>();
+            calllbacksComponent.view = this;
+
```
Comment (Graicc) [inline ComputerInterface/Views/GameSettings/JoinRoomView.cs:33]: Why did you move this? You shouldn't need to create a new GameObject every time

Comment (developer9998) [reply ComputerInterface/Views/GameSettings/JoinRoomView.cs:33]: Not 100% but I think it was causing some issues?

**Comment by Graicc** [comment]
The `ComputerSettingsView` can just ask for the `CustomComputer` because Zenject will ensure that it gets it when its constructed by the `ComputerViewFactory`, which avoids the need to have a singleton (this is basically the point of dependency injection)

**Comment by Graicc** [comment]
LGTM
